generator/src/main/java/dev/cdevents/generator/CDEventsGenerator.java

```diff
@@ -153,15 +154,29 @@ private static void updateSubjectContentProperties(SchemaData schemaData, JsonNo
             String capitalizedContentField = StringUtils.capitalize(contentField);
             JsonNode contentNode = contentMap.getValue();
             String dataType = "";
-            if (!contentNode.get("type").asText().equals("object")) {
+            if (contentNode.get("type") != null && !contentNode.get("type").asText().equals("object")) {
```

Comment:
I wonder if we could use the visitor pattern in the future here. It may clean up the code quite a bit.